### PR TITLE
network: provide exceptions with more info

### PR DESCRIPTION
**Pull request by thc202**
Change network add-on to throw exceptions with more information, for timeouts include its value and for unknown host names whether it was while resolving the proxy's host.

### CODE REVIEW

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/apachev5/HttpSenderApache.java [hunk 1]
```diff
@@ -292,6 +296,12 @@ protected void sendImpl(
             throws IOException {
         try {
             sendImpl0(ctx, requestContext, message, responseBodyConsumer);
+        } catch (SocketTimeoutException e) {
+            LOGGER.debug("A timeout occurred while sending the request:", e);
+            throw new ZapSocketTimeoutException(e.getMessage(), options.getTimeoutInSecs());
+        } catch (UnknownHostException e) {
+            LOGGER.debug("An unknown host occurred while sending the request:", e);
```
Comment (kingthorin) [inline addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/apachev5/HttpSenderApache.java:303]: Add "exception" or re-word as some sort of DNS resolution failure?

**Comment by kingthorin** [comment]
```console
> Task :addOns:selenium:testClasses


1601 tests completed, 4 failed
> Task :addOns:network:test

> Task :addOns:network:test FAILED
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended

> Task :addOns:selenium:test


FAILURE: Build failed with an exception.
```

**Comment by kingthorin** [comment]
Oops, sorry about the close, hit the wrong button.

**Review by kingthorin — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thanks!

**Pull request merged**